cmd/customcli/main.go

```diff
@@ -1,5 +1,77 @@
 package main
 
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+// commands is a register of all availables commands that can be executed by
+// this program. The name is used to match with the first argument given.
+//
+// When a cmd function is callend it is given stdin, stdout and command line
+// arguments except the program name and this command name. It is the
+// responsibility of the command function to parse the arguments. Use os.Stderr
+// to write error messages.
+//
+// A command function is an independent runable that is taking input and output
+// being stdin and stdout. Given args are the command line arguments, without
+// the program name, that should be parsed using the flag package.
+// A command function is expected to read and write only to provided input and
+// output. In a special case of an invalid argument a message to os.Stderr and
+// os.Exit(2) call are allowed.
+//
+// When implementing a command function, keep it simple. A command function
+// should provide a single functionality. A unix pipe can be used to construct
+// a pipeline. For example, there are 3 separate functions for creating a
+// transaction, signing and submitting. They can be combined into a single
+// pipeline line:
+//
+//   $ customcli release-escrow -escrow 1 \
+//       | customcli as-proposal \
+//       | customcli sign \
+//       | customcli submit
+//
+var commands = map[string]func(input io.Reader, output io.Writer, args []string) error{}
```

Comment:
empty `commands` might be confusing. Maybe add a comment as a body explaining how to register functions.